src/test/java/org/candlepin/subscriptions/security/IdentityHeaderAuthenticationDetailsServiceTest.java

```diff
@@ -60,6 +64,13 @@ void testAdminRoleGranted() throws Exception {
     assertRoles(false, RoleProvider.SWATCH_ADMIN_ROLE);
   }
 
+  @Test
+  void testReportReaderRoleGranted() throws RbacApiException {
```

Comment:
The semantics here are a little strange.  You have an assertion of false, but your test name suggests you're testing which roles were granted.  Either rename it, explain it in comments, or add an assertion concerning which rol(s) you expect to see.

Reply:
The assertion isn't false; it's the parameter to `extractRoles` that's false.  The assertion is "assert that the returned collection from extractRoles contains SWATCH_REPORT_READER."